src/ring_theory/perfection.lean

```diff
@@ -0,0 +1,366 @@
+/-
+Copyright (c) 2020 Kenny Lau. All rights reserved.
+Released under Apache 2.0 license as described in the file LICENSE.
+Authors: Kenny Lau
+-/
+
+import algebra.char_p
+import algebra.ring.pi
+import analysis.special_functions.pow
+import field_theory.perfect_closure
+import ring_theory.localization
+import ring_theory.subring
+import ring_theory.valuation.integers
+
+/-!
+# Ring Perfection and Tilt
+
+In this file we define the perfection of a ring of characteristic p, and the tilt of a field
+given a valuation to `ℝ≥0`.
+
+## TODO
+
+Define the valuation on the tilt, and define a characteristic predicate for the tilt.
+
+-/
+
+universes u v w
```

Comment:
Later on, you'll use `v` for a valuation. I think it helps avoid confusion if we don't use it for a universe variable as well.

Reply:
Done